test/cli.test.js

```diff
@@ -4,301 +4,342 @@ process.env.TZ = 'UTC'
 
 const path = require('node:path')
 const { spawn } = require('node:child_process')
-const { test } = require('tap')
+const { test } = require('node:test')
+const { once } = require('./helper')
 
 const bin = require.resolve(path.join(__dirname, '..', 'bin.js'))
 const epoch = 1522431328992
 const logLine = '{"level":30,"time":1522431328992,"msg":"hello world","pid":42,"hostname":"foo"}\n'
 const env = { TERM: 'dumb', TZ: 'UTC' }
 const formattedEpoch = '17:35:28.992'
 
-test('cli', (t) => {
-  t.test('does basic reformatting', (t) => {
+test('cli', async (t) => {
+  await t.test('does basic reformatting', async (t) => {
     t.plan(1)
     const child = spawn(process.argv[0], [bin], { env })
-    child.on('error', t.threw)
-    child.stdout.on('data', (data) => {
-      t.equal(data.toString(), `[${formattedEpoch}] INFO (42): hello world\n`)
+    child.on('error', t.assert.fail)
+    const endPromise = once(child.stdout, 'data', (data) => {
+      t.assert.strictEqual(data.toString(), `[${formattedEpoch}] INFO (42): hello world\n`)
     })
     child.stdin.write(logLine)
-    t.teardown(() => child.kill())
+    await endPromise
+    t.after(() => child.kill())
   })
 
-  ;['--levelFirst', '-l'].forEach((optionName) => {
-    t.test(`flips epoch and level via ${optionName}`, (t) => {
-      t.plan(1)
-      const child = spawn(process.argv[0], [bin, optionName], { env })
-      child.on('error', t.threw)
-      child.stdout.on('data', (data) => {
-        t.equal(data.toString(), `INFO [${formattedEpoch}] (42): hello world\n`)
+  await Promise.all(['--levelFirst', '-l']
+    .map(async (optionName) => {
+      await t.test(`flips epoch and level via ${optionName}`, async (t) => {
+        t.plan(1)
+        const child = spawn(process.argv[0], [bin, optionName], { env })
+        child.on('error', t.assert.fail)
+        const endPromise = once(child.stdout, 'data', (data) => {
+          t.assert.strictEqual(data.toString(), `INFO [${formattedEpoch}] (42): hello world\n`)
+        })
+        child.stdin.write(logLine)
+        await endPromise
+        t.after(() => child.kill())
```

Comment:
I think it is easier to read code like:

```js
for (const option of ['a', 'b']) {
	await t.test('something', async t => {
		// do test
	})
}
```

The `Promise.all` and `.map` obscure things to me.

Reply:
Done